src/parser.rs

```diff
@@ -1735,16 +1834,19 @@ impl<'a> Parser<'a> {
         statements: &mut Vec<Statement>,
         until: Statement,
     ) {
+        let mut out = Vec::new();
         while let Some(statement_raw) = iterator.next() {
             if let Statement::Raw(line_raw) = statement_raw {
                 let statement = self.parse_statement(line_raw, constructor, stack, iterator);
                 if statement == until {
                     break
                 } else {
-                    statements.push(statement)
+                    out.push(statement)
```

Comment:
better to check here if the new statement is a block (catch, else if, else, etc.) and last statement in out is a comment, then we will take out the last statement, append the new statement and add the comments to the beginning of the new statement, that way we do not cycle through all statements again